packages/common/src/store/buy-usdc/sagas.ts

```diff
@@ -0,0 +1,215 @@
+    mint: 'usdc',
+    tokenAccount,
+    initialBalance,
+    retryDelayMs,
+    maxRetryCount
+  })
+
+  // Check that we got the requested amount
+  const purchasedAmount = new BN(newBalance).sub(new BN(initialBalance))
+  if (purchasedAmount !== new BN(desiredAmount)) {
+    console.warn(
+      `Warning: Purchase USDC amount differs from expected. Actual: ${new BN(
+        newBalance
+      )
+        .sub(new BN(initialBalance))
+        .toNumber()} Wei. Expected: ${desiredAmount / 100} USDC.`
+    )
+  }
+
+  return { newBalance }
+}
+
+function* doBuyUSDC({
+  payload: { desiredAmount }
+}: ReturnType<typeof onRampOpened>) {
+  const reportToSentry = yield* getContext('reportToSentry')
+  const userBank = yield* getUSDCUserBank()
+  const { track, make } = yield* getContext('analytics')
+  const provider = yield* select(getBuyUSDCProvider)
+
+  try {
+    if (provider !== USDCOnRampProvider.STRIPE) {
+      throw new Error('USDC Purchase is only supported via Stripe')
+    }
+
+    yield* put(
+      initializeStripeModal({
+        // stripe expects amount in dollars, not cents
+        amount: (desiredAmount / 100).toString(),
+        destinationCurrency: 'usdc',
+        destinationWallet: userBank.toString(),
+        onRampCanceled,
+        onRampSucceeded
+      })
+    )
+
+    yield* put(setVisibility({ modal: 'StripeOnRamp', visible: true }))
+
+    // Record start
+    yield* call(
+      track,
+      make({ eventName: Name.BUY_USDC_ON_RAMP_OPENED, provider })
+    )
+
+    // Setup
+
+    // Get config
+    const { retryDelayMs, maxRetryCount } = yield* call(getBuyUSDCRemoteConfig)
+
+    // Wait for purchase
+    // Have to do some typescript finangling here due to the "race" effect in purchaseStep
+    // See https://github.com/agiledigital/typed-redux-saga/issues/43
+    const { newBalance } = (yield* call(purchaseStep, {
+      provider,
+      desiredAmount,
+      tokenAccount: userBank,
+      retryDelayMs,
+      maxRetryCount
+    }) as unknown as ReturnType<typeof purchaseStep>)!
+
+    // If the user canceled the purchase, stop the flow
+    if (newBalance === undefined) {
+      return
+    }
+
+    yield* put(buyUSDCFlowSucceeded())
+
+    // Record success
+    yield* call(
+      track,
+      make({
+        eventName: Name.BUY_USDC_SUCCESS,
+        provider,
+        requestedAmount: desiredAmount
+      })
+    )
+  } catch (e) {
+    yield* call(reportToSentry, {
+      level: ErrorLevel.Error,
+      error: e as Error,
+      additionalInfo: { userBank }
+    })
+    yield* put(buyUSDCFlowFailed())
+    yield* call(
+      track,
+      make({
+        eventName: Name.BUY_USDC_FAILURE,
+        provider,
+        requestedAmount: desiredAmount,
+        error: (e as Error).message
+      })
+    )
+  }
+}
+
+function* doStartBuyUSDCFlow(action: ReturnType<typeof startBuyUSDCFlow>) {
+  // This is a placeholder action to handle a general buy usdc flow.
```

Comment:
I know this mirrors `BuyAudio`, but in this case I don't think we have an amount selector modal or anything (since we're decoupling) so we might be able to cut the `startBuyUSDCFlow` here and just use `onRampOpened`

Reply:
Fair point. I had imagined we might allow ad-hoc purchase of USDC where you can select an amount first. I can remove this for now.

Reply:
I think even in that case we can have the new modal call `onRampOpened` directly

Reply:
Removed the start action and refactored a little bit so that the purchase content flow calls the onramp action directly.